client/components/common/BuildList.tsx

```diff
@@ -96,6 +117,51 @@ const BuildList: React.FC<Props> = ({
     DEBOUNCE_INTERVAL,
   );
 
+  // Takes params from the router and filters them out so that the inherent username value is not in the address bar
+  // twice, also when values are undefined or empty strings. Returns a new object to use in formatQueryString.
+  const formatParams = (obj: ProfileQueryParams) =>
+    Object.entries(obj)
+      .filter(([, val]) => val !== undefined && val !== '' && val !== username)
+      .reduce((result, [key, val]) => {
+        const newResult = {
+          ...result,
+          [key]: val,
+        };
+        return newResult;
+      }, {} as ProfileQueryParams);
+
+  const formatQueryString = () => {
```

Comment:
Why does this need to be a function?

Reply:
The original idea is separating the big function from the smaller replace method on the router, but I don't mind either way, could move the logic to the replace and make it an anonymous function.